ext/gd/config.m4

```diff
@@ -171,33 +185,34 @@ int main(int argc, char** argv) {
   [AS_VAR_SET([php_var], [yes])],
   [AS_VAR_SET([php_var], [no])],
   [AS_VAR_SET([php_var], [no])])])
-AS_VAR_IF([php_var], [yes],
-  [AC_DEFINE_UNQUOTED([$2], [1],
-    [Define to 1 if GD library has 'gdImageCreateFrom$1'.])])
+AS_VAR_IF([php_var], [yes], [$2])
 AC_LANG_POP([C])
 CFLAGS="${old_CFLAGS}"
 LIBS="${old_LIBS}"
 AS_VAR_POPDEF([php_var])
 ])
 
 AC_DEFUN([PHP_GD_CHECK_VERSION],[
-  PHP_GD_CHECK_FORMAT([Png],  [HAVE_GD_PNG])
-  PHP_GD_CHECK_FORMAT([Avif], [HAVE_GD_AVIF])
-  PHP_GD_CHECK_FORMAT([Webp], [HAVE_GD_WEBP])
-  PHP_GD_CHECK_FORMAT([Jpeg], [HAVE_GD_JPG])
-  PHP_GD_CHECK_FORMAT([Xpm],  [HAVE_GD_XPM])
-  PHP_GD_CHECK_FORMAT([Bmp],  [HAVE_GD_BMP])
-  PHP_GD_CHECK_FORMAT([Tga],  [HAVE_GD_TGA])
+  PHP_GD_CHECK_FORMAT([Png],  [AC_DEFINE([HAVE_GD_PNG], [1])])
+  PHP_GD_CHECK_FORMAT([Avif], [AC_DEFINE([HAVE_GD_AVIF], [1])])
+  PHP_GD_CHECK_FORMAT([Webp], [AC_DEFINE([HAVE_GD_WEBP], [1])])
+  PHP_GD_CHECK_FORMAT([Jpeg], [AC_DEFINE([HAVE_GD_JPG], [1])])
+  PHP_GD_CHECK_FORMAT([Xpm],  [AC_DEFINE([HAVE_GD_XPM], [1])])
+  PHP_GD_CHECK_FORMAT([Bmp],  [AC_DEFINE([HAVE_GD_BMP], [1])])
+  PHP_GD_CHECK_FORMAT([Tga],  [AC_DEFINE([HAVE_GD_TGA], [1])])
   PHP_CHECK_LIBRARY([gd], [gdFontCacheShutdown],
-    [AC_DEFINE([HAVE_GD_FREETYPE], [1], [ ])],
+    [AC_DEFINE([HAVE_GD_FREETYPE], [1])],
     [],
     [$GD_SHARED_LIBADD])
   PHP_CHECK_LIBRARY([gd], [gdVersionString],
-    [AC_DEFINE([HAVE_GD_LIBVERSION], [1], [ ])],
+    [AC_DEFINE([HAVE_GD_LIBVERSION], [1],
+      [Define to 1 if GD library has the 'gdVersionString' function.])],
     [],
     [$GD_SHARED_LIBADD])
   PHP_CHECK_LIBRARY([gd], [gdImageGetInterpolationMethod],
-    [AC_DEFINE([HAVE_GD_GET_INTERPOLATION], [1], [ ])],
+    [AC_DEFINE([HAVE_GD_GET_INTERPOLATION], [1],
+      [Define to 1 if GD library has the 'gdImageGetInterpolationMethod'
+      function.])],
```

Comment:
Nice. `gdImageGetInterpolationMethod()` is available as of libgd 2.1.1; do we still support such old versions?

However, not really related to this PR; just mentioning.

Reply:
can be merged as is however centos 8 libgd version is 2.2.5. can be addressed in another PR eventually.

Reply:
I'm all in favor of bumping the libgd minimum version however issue is that bundled libgd is of version ~2.0.35. Probably adjusted already to a partially newer version. But it doesn't have the gdImageGetInterpolationMethod as an exported function yet.

Reply:
> bundled libgd is of version ~2.0.35

Yeah, that is bad. There is special support for this, so it claims to be 2.1.0 compatible, but that is also wrong. In fact it is closer to 2.2 or likely even 2.3, although it is certainly not fully compatible with any system libgd. Long time ago I suggested to update the bundled GD version numbers/string, but that was rejected because the goal was to drop bundled GD – but that still didn't happen.

Anyway, we can resolve this in another PR, and just ignore this for this PR.